botorch/sampling/pathwise/prior_samplers.py

```diff
@@ -102,36 +117,108 @@ def _draw_kernel_feature_paths_ExactGP(
 ) -> GeneralizedLinearPath:
     (train_X,) = get_train_inputs(model, transformed=False)
     return _draw_kernel_feature_paths_fallback(
-        num_inputs=train_X.shape[-1],
         mean_module=model.mean_module,
         covar_module=model.covar_module,
         input_transform=get_input_transform(model),
         output_transform=get_output_transform(model),
         is_ensemble=is_ensemble(model),
+        num_ambient_inputs=train_X.shape[-1],
         **kwargs,
     )
 
 
-@DrawKernelFeaturePaths.register(ModelListGP)
-def _draw_kernel_feature_paths_list(
-    model: ModelListGP,
-    join: Callable[[list[Tensor]], Tensor] | None = None,
+@DrawKernelFeaturePaths.register(models.ModelListGP)
+def _draw_kernel_feature_paths_ModelListGP(
+    model: models.ModelListGP,
+    reducer: Callable[[List[Tensor]], Tensor] | None = None,
     **kwargs: Any,
 ) -> PathList:
     paths = [draw_kernel_feature_paths(m, **kwargs) for m in model.models]
-    return PathList(paths=paths, join=join)
+    return PathList(paths=paths, reducer=reducer)
+
+
+@DrawKernelFeaturePaths.register(models.MultiTaskGP)
+def _draw_kernel_feature_paths_MultiTaskGP(
+    model: models.MultiTaskGP, **kwargs: Any
+) -> GeneralizedLinearPath:
+    (train_X,) = get_train_inputs(model, transformed=False)
+    num_ambient_inputs = train_X.shape[-1]
+    task_index = (
+        num_ambient_inputs + model._task_feature
+        if model._task_feature < 0
+        else model._task_feature
+    )
+
+    # Extract kernels from the product kernel structure
+    # model.covar_module is a ProductKernel
+    # containing data_covar_module * task_covar_module
+    from gpytorch.kernels import ProductKernel
+
+    if isinstance(model.covar_module, ProductKernel):
```

Comment:
The MTGPs (including the fully Bayesian variant) now use a ProductKernel by definition, so this check and the conditional logic should be obsolete.

Reply:
And if there are some edge cases where we don't adhere to the standard `ProductKernel(IndexKernel, SomeOtherKernel)` where both have the expected `active_dims`, I think it's best to raise an error!